search/segment_tree.hpp

```diff
@@ -66,9 +70,16 @@ class SegmentTree
 
   // Calls |fn| on arbitrary segment containing |x|.
   template <typename Fn>
-  void Find(double x, Fn && fn) const
+  void FindAny(double x, Fn && fn) const
   {
-    return Find(0 /* index */, x, fn);
+    FindAny(0 /* index */, x, fn);
+  }
+
+  // Calls |fn| on all segments containing |x|.
+  template <typename Fn>
+  void FindAll(double x, Fn && fn) const
+  {
+    FindAll(0 /* index */, x, fn);
```

Comment:
forward?

Reply:
No need to forward here. Implementations are recursive.